sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/EventGridPublisherAsyncClient.java

```diff
@@ -70,17 +74,26 @@ public EventGridServiceVersion getServiceVersion() {
      * @return the completion.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Void> sendEvents(Iterable<EventGridEvent> events) {
+    public Mono<Void> sendEventGridEvents(Iterable<EventGridEvent> events) {
         if (events == null) {
             return monoError(logger, new NullPointerException("'events' cannot be null."));
         }
-        return withContext(context -> sendEvents(events, context));
+        return withContext(context -> sendEventGridEvents(events, context));
     }
 
-    Mono<Void> sendEvents(Iterable<EventGridEvent> events, Context context) {
+    Mono<Void> sendEventGridEvents(Iterable<EventGridEvent> events, Context context) {
         final Context finalContext = context != null ? context : Context.NONE;
         return Flux.fromIterable(events)
-            .map(EventGridEvent::toImpl)
+            .map(event -> {
+                com.azure.messaging.eventgrid.implementation.models.EventGridEvent internalEvent = event.toImpl();
```

Comment:
This logic is at two places (also at line 121) Can we combine it in a function ?

Reply:
They look the same but one is for EventGridEvent and one for CloudEvent. the two classes don't have a common parent class or interface so I prefer not combining these lines of code.
Good points though.